src/Surfnet/StepupGateway/SamlStepupProviderBundle/Controller/SamlProxyController.php

```diff
@@ -161,7 +161,17 @@ public function consumeAssertionAction($provider, Request $httpRequest)
                 )
             );
         } catch (SecondfactorVerfificationRequiredException $e) {
-            return $this->forward('SurfnetStepupGatewayGatewayBundle:SecondFactor:gssfVerified');
+            // The provider state handler has no access to the session object, hence we use the proxy state handler
```

Comment:
Could you elaborate what the provider state handler is and what the difference is between that and the proxy state handler?

Reply:
The provider state handler is the GSSP state handler. Which does not utilize session state.